clang/lib/Sema/SemaAccess.cpp

```diff
@@ -39,7 +39,8 @@ bool Sema::SetMemberAccessSpecifier(NamedDecl *MemberDecl,
                                     AccessSpecifier LexicalAS) {
   if (!PrevMemberDecl) {
     // Use the lexical access specifier.
-    MemberDecl->setAccess(LexicalAS);
```

Comment:
@shafik https://github.com/llvm/llvm-project/issues/112208#issuecomment-2411881773 when `PrevMemberDecl` exists but isn't equal to `LexicalAS` — with `AS_none` excluded

https://github.com/llvm/llvm-project/blob/020566701030425f44eb80387d0ae76c5a867aa9/clang/lib/Sema/SemaAccess.cpp#L48

 I suppose `AS_none` should be excluded when `PrevMemberDecl` is unknown. What do you think?

Reply:
I feel like we should be catching this in `Sema::ActOnTag` here:

https://github.com/llvm/llvm-project/blob/b49701085414838deb0213b9f10b68d9d3af6b0a/clang/lib/Sema/SemaDecl.cpp#L17938-L17951

It looks like we are not b/c `TUK` is `TUK_Declataion` but we should not be forward declaring here either. 

The current diagnostic does not fit this case though and maybe it has to be tweaked some more.

CC @AaronBallman

Reply:
Please also add more details to the summary, it is helpful to have some idea what you think the underlying problem is and why this fix is correct before reviewing.

Reply:
@shafik could you take a look at the latest changes? thanks

Reply:
I think these changes make sense, the diagnostic on forward reference is not great but that feels like a seperate already existing problem and not directly related to the crash.

Please update the summary. I also had one question on why we are checking `Name` as well.

Reply:
@shafik thanks for the feedback. I've checked, and it seems safe to omit `Name` here, so I’ve removed it.